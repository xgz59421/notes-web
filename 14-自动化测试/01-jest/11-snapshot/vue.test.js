// 引入完整版vue
import Vue from 'vue/dist/vue'

function renderVueComponent() {
  document.body.innerHTML = `<div id='app'></div>`
  new Vue({
    template: `
    <div id='app'>
      <h1>{{msg}}</h1>
    </div>`,
    data: {
      msg: 'hello'
    }
  }).$mount('#app')
}

test('snapshot test', () => {
  renderVueComponent()
  // 第1次运行的时候, 会生成快照文件字符串
  // 下一次运行测试的时候会和快照文件进行比对, 如果不一致则测试失败
  expect(document.body.innerHTML).toMatchSnapshot()
})