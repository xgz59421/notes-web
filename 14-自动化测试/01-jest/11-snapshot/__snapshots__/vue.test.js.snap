// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `"<div id=\\"app\\"><h1>hello</h1></div>"`;
